src/rimecandidate.cpp

```diff
@@ -138,4 +150,50 @@ const CandidateWord &RimeCandidateList::candidateFromAll(int idx) const {
 }
 
 int RimeCandidateList::totalSize() const { return -1; }
+
+void RimeCandidateList::setActionableImpl(
+    std::unique_ptr<ActionableCandidateList> actionable) {
+    actionable_ = std::move(actionable);
```

Comment:
There is no need to do this. Since RimeCandidateList itself is already a class that is defined in our source, you can subclass ActionableCandidateList, and call setActionable(this).